cmd/cfssl/cfssl_test.go

```diff
@@ -1 +1,480 @@
+	}
+	// Checks for correctness between API and CLI responses
+	if !reflect.DeepEqual(infoAPIResp.Result.Certificate,
+		infoCLIResp.Certificate) {
+		t.Fatalf("Info API Certificate did not match with CLI")
+	}
+}
+
+// Below are API endpoints without a CLI counterpart.
+// They are currently being checked for just successful output,
+// not necessarily matching any existing CLI output.
+func TestScanInfo(t *testing.T) {
+	response := apiOut("GET", nil, postStr+"scaninfo", t)
+	scaninfoAPIResp := responseUnmarshal(response)
+	if !scaninfoAPIResp.Success {
+		t.Fatalf("ScanInfo API Failed Response")
+	}
+}
+
+func TestInitCA(t *testing.T) {
+	response := apiOut("POST", hosts, postStr+"init_ca", t)
+	initcaAPIResp := responseUnmarshal(response)
+	if !initcaAPIResp.Success {
+		t.Fatalf("InitCA API Failed Response")
+	}
+}
+
+func TestNewCert(t *testing.T) {
+	jsonStr := []byte(`{ "request": ` + string(newcertHosts) + ` }`)
+	response := apiOut("POST", jsonStr, postStr+"newcert", t)
+	fmt.Println(string(response))
+	newcertAPIResp := responseUnmarshal(response)
+	if !newcertAPIResp.Success {
+		t.Fatalf("NewCert API Failed Response")
+	}
+}
+
+func TestNewKey(t *testing.T) {
+	response := apiOut("POST", hosts, postStr+"newkey", t)
+	newkeyAPIResp := responseUnmarshal(response)
+	if !newkeyAPIResp.Success {
+		t.Fatalf("NewKey API Failed Response")
+	}
+}
+
+func TestAuthSign(t *testing.T) {
+	// random key string generated from open('/dev/random').read(32).encode('hex')
+	authkey := "f66761a3098baaa7c893412d54557b8ff22e69eb4f26f958b769ff6ff7f7271b"
+
+	csr := []byte(`{
+	    "hosts": [
+	        "127.0.0.1"
+	    ],
+	    "certificate_request": "-----BEGIN CERTIFICATE REQUEST-----\n` +
+		`MIHyMIGaAgEAMBYxFDASBgNVBAMTC3Rlc3Qgc2VydmVyMFkwEwYHKoZIzj0CAQYI\n` +
+		`KoZIzj0DAQcDQgAENosvCA8k6wJNm16fX2CJlh105I5kJrTCktsDfFKTtejqRQy5\n` +
+		`Ypx7mfI7hR5dLxdj9H0WXO/1i/mmoBENYm/GJ6AiMCAGCSqGSIb3DQEJDjETMBEw\n` +
+		`DwYDVR0RBAgwBocEfwAAATAKBggqhkjOPQQDAgNHADBEAiBjTbRpcdnSXoj32HIM\n` +
+		`mG10OjtcpHPmejabVQBpI3hWxwIgQFqGwGYyf2HfYhTBZUs93GlQqqoeWkvNV4I2\n` +
+		`2iSfOek=\n-----END CERTIFICATE REQUEST-----\n",
+	    "profile": "server",
+	    "label": ""
+	}`)
+
+	testProvider, _ := auth.New(authkey, nil)
+
+	token, _ := testProvider.Token(csr)
+	request := base64.StdEncoding.EncodeToString(csr)
+
+	jsonStr := []byte(`{
+	    "timestamp": 1447094758,
+	    "token": "` + base64.StdEncoding.EncodeToString(token) + `",
+	    "request": "` + request + `"
+    }`)
+
+	req, _ := http.NewRequest("POST", postStr+"authsign", bytes.NewBuffer(jsonStr))
+	client := &http.Client{}
+
+	// Start up server + Execute request
+	cmd := exec.Command("cfssl", "serve", "-address=127.0.0.1", "-port=8888",
+		"-ca=testdata/cert.pem", "-ca-key=testdata/cert-key.pem",
+		"-config=testdata/config.json")
+	if err := cmd.Start(); err != nil {
+		t.Fatalf("Error running cfssl serve, %v", err)
+	}
+
+	// Give the server time to start up
+	time.Sleep(500 * time.Millisecond)
```

Comment:
I ran into an issue with the RO API tests where the sleep time wasn't enough for slower computers. Might be helpful to test this out on another computer